### PR TITLE
add a section to plugin manifest for kernel to be modprobed before en…

### PR DESCRIPTION
**Pull request by vieux**
+cleanup documentation

ping @anusha-ragunathan @tiborvass

### CODE REVIEW

**Comment by vieux** [comment]
ping @anusha-ragunathan

File: api/types/plugin.go [hunk 1]
```diff
@@ -127,6 +128,12 @@ type PluginArgs struct {
 	Value []string
 }
 
+// PluginKernelModule represents a Loadable Kernel Modules (LKM) to modprobe before enabling a plugin
+type PluginKernelModule struct {
+	PluginSetting
```
Comment (anusha-ragunathan) [inline api/types/plugin.go]: What is the need for `PluginSetting` here? It doesnt seem to be used anywhere in the code.

**Comment by anusha-ragunathan** [comment]
Most importantly, Kernel Modules should *always* ask user permission. It should be part of ComputePrivileges. https://github.com/docker/docker/blob/master/plugin/v2/plugin_experimental.go#L124

File: api/types/plugin.go (hunk 1 above)
Comment (vieux) [inline api/types/plugin.go]: it's never used for now right ? my understanding was at some point `docker plugin set` will allow the user to change the modules, but maybe it shouldnt ?

Comment (anusha-ragunathan) [inline api/types/plugin.go]: I dont think it should be settable by user.

**Comment by vieux** [comment]
@anusha-ragunathan PTAL

**Comment by anusha-ragunathan** [comment]
@vieux : Can you provide output when you tested this?

File: docs/extend/manifest.md [hunk 2]
```diff
@@ -50,11 +50,11 @@ Manifest provides the base accessible fields for working with V0 plugin format
       
     - **`types`** *string array*
 
-      types indicate what interface(s) the plugin currently implements.
+          types indicate what interface(s) the plugin currently implements.
```
Comment (thaJeztah) [inline docs/extend/manifest.md]: This document is now published from https://github.com/docker/docker.github.io, so this change should be reverted, and reopened as a PR in that repository

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by cpuguy83** [comment]
What's the use-case for this?
If the plugin requires a kernel module, presumably they will have to have compiled the module for the kernel. What utility is there at that point in probing for the module?

**Comment by anusha-ragunathan** [comment]
@cpuguy83 : The use case is, when the plugin requires modprobing a kernel module that is part of the standard Linux kernel. For example, open-iscsi is the standard iSCSI initiator and the plugin would require it to modprobed, to be functional. 
This is not intended to download and/or compile the module for the underlying host kernel. It is strictly to modprobe a module that exists across all kernels. If the modprobe fails, the plugin install will fail.

**Comment by anusha-ragunathan** [comment]
It's a very corner case and very low in priority. Lets close this now and reopen if there are more such usecases

**Comment by justincormack** [comment]
Generally kernel modules autoload, I don't think you need to manually modprobe them very often at all.

File: docs/extend/manifest.md (hunk 2 above)
Comment (thaJeztah) [inline docs/extend/manifest.md]: @vieux this was reconsidered, so can remain in this PR

**Comment by thaJeztah** [comment]
Closing, as this was reopened by mistake, and it's no longer needed

**Pull request closed without merging**